go/interval/interval.go

```diff
@@ -0,0 +1,111 @@
+package interval
+
+import (
+	"errors"
+	"sync"
+	"time"
+)
+
+type scheduled interface {
+	nextRun() (time.Duration, error)
+}
+
+type Job struct {
+	fn        func()
+	Quit      chan bool
+	SkipWait  chan bool
+	err       error
+	schedule  scheduled
+	isRunning bool
+	sync.RWMutex
+}
+
+type recurrent struct {
+	delay   int64
+	started time.Time
+	count   int64
+	done    bool
+}
+
+func (r *recurrent) nextRun() (time.Duration, error) {
+	if r.delay == 0 {
+		return 0, errors.New("invlaid delay")
+	}
+	if !r.done {
+		r.done = true
+		return 0, nil
+	}
+	//adjust offset for processing time drift
+	offset := (r.count * r.delay) - time.Since(r.started).Nanoseconds()
+	return time.Duration(r.delay + offset), nil
+}
+
+func Every(duration string) *Job {
+	t, err := time.ParseDuration(duration)
+	if err != nil {
+		return &Job{err: errors.New("Could not parse duration: " + duration)}
```

Comment:
The golang way would be to return (*Job, error) and not have errors in the job struct

Anyway the external caller can't even access err because its private

Reply:
It's been adjusted. The error returns fine w/ the given example from the `README`

Reply:
Still not our standard way to do it though

And consumers from other packages won't be able to access err so it should definitely be put as a return argument instead